### PR TITLE
fix to handle chained contains query

### PR DESCRIPTION
**Pull request by v-mserdar**
## Description
This revision will handle the scenario that previously caused the generated sql to incorrectly join on the wrong columns. See the related item for the specifics.

## Related issues
Addresses [issue #97812](https://microsofthealth.visualstudio.com/Health/_workitems/edit/97812).

## Testing
I ran all tests as well as created a new test to handle this scenario

## FHIR Team Checklist
- [x] **Update the title** of the PR to be succinct and less than 50 characters
- [ ] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [x] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [ ] Tag the PR with **Azure API for FHIR** if this will release to the Azure API for FHIR managed service (CosmosDB or common code related to service)
- [x] Tag the PR with **Azure Healthcare APIs** if this will release to the Azure Healthcare APIs managed service (Sql server or common code related to service)
- [ ] CI is green before merge [Build Status](https://microsofthealthoss.visualstudio.com/FhirServer/_build/latest?definitionId=27&branchName=main) 
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Tests.Common/Microsoft.Health.Fhir.Tests.Common.csproj [hunk 1]
```diff
@@ -6,6 +6,9 @@
     <AllowedOutputExtensionsInPackageBuildOutputFolder>$(AllowedOutputExtensionsInPackageBuildOutputFolder);.pdb</AllowedOutputExtensionsInPackageBuildOutputFolder>
     <IsTestProject>false</IsTestProject>
   </PropertyGroup>
+  <ItemGroup>
```
Comment (fhibf) [inline src/Microsoft.Health.Fhir.Tests.Common/Microsoft.Health.Fhir.Tests.Common.csproj:9]: Curious why this is the only file under "remove".

Comment (v-mserdar) [reply src/Microsoft.Health.Fhir.Tests.Common/Microsoft.Health.Fhir.Tests.Common.csproj]: Fixed in latest commit

**Review by fhibf — APPROVED** [review]

**Pull request merged**